### ISSUE TITLE
[Feedback] GitHub Project boards workflow

### ISSUE THREAD

**Issue by choldgraf** [2i2c-org/team-compass#182]
# Summary

We've been trying out a workflow recently based around deliverables and project boards. [You can find a description of this workflow here](https://team-compass.2i2c.org/en/latest/practices/development.html#project-backlogs).

As we try out this workflow, we should think about ways that we can improve these processes, structure the boards, etc. Let's leave this issue for a month or two to collect feedback and iterate on ideas, and we can close it once we are happy with the process.

# Actions

- [ ] Collect feedback over a few weeks of running this meeting
- [ ] Incorporate feedback over time or in focused development
- [ ] Close this after a few cycles

**Comment by choldgraf** [comment]
Some thoughts from the last deliverables sync meeting and recent conversation with @GeorgianaElena:

- [ ] Frame the activity board in terms of "actions that 2i2c team members should take"
- [ ] Separate out "blocked and waiting" columns
  - Blocked means that a team member should help un-block somebody
  - Waiting is something where a non-2i2c person needs to take action
  - (@yuvipanda I think you're the one that described this, let me know if this isn't quite right)
- [ ] The "info" column could be a place to store "ongoing information and reminders" about things like:
  - Who is the current support steward?
  - Important upcoming dates
  - Issues that ask for ongoing feedback from everybody and won't be closed with a single action
- [ ] Reframe the difference between "deliverables" boards and "activity" boards to be about complexity rather than about "deliverable vs. task". Tasks can be complex and multi-step as well, so this can be a confusing breakdown.
  - One rubric: "deliverables generally take > 3 days to complete, tasks are finished in 0-3 days.
  - Another rubric: "deliverables are often made up of many steps or components to accomplish. Each step should be on the activity board, and the high-level issue should be in the deliverables board".
  - Another rubric: "the activity board should have items that are close-able with a single, relatively straightforward PR"
- [ ] In general we should put **Issues on the project boards**, not direct links to PRs. If you've got a PR, link it to the issue so that it will close the issue when merged. If this is done, then a link to the PR will automatically show up for that item in the Activity board.

@GeorgianaElena let me know if I missed something.

**Comment by choldgraf** [comment]
- [ ] Have two "Done" columns -
  - a regular "Done" column
  - a "Done Archive"

Before every team deliverables review meeting we move all "Done" cards into the "Done archive" (I believe we can automate this)

**Comment by yuvipanda** [comment]
I'm currently a little overwhelmed by the boards, and want to move around the process a little to help with that. Specifically, I was wondering if we could:

1. Unify to one board, not two
2. Keep the level of items in the board very high, so there's not too many of them. Currently almost all the columns in all the boards have too many items for my brain
3. Use more GItHub native features (like assignment and reviewer requests) over using a board to signal that information via columns

**Comment by choldgraf** [comment]
We discussed this in a recent team meeting, and agreed on the changes that are encoded in this PR: https://github.com/2i2c-org/team-compass/pull/188

**Comment by damianavila** [comment]
@choldgraf, thanks for putting into words the ideas we have been discussing recently.

Regarding the last item, I am wondering if, at this time, the Backlog Steward and Sprint Steward should be the same person since the information coming from the backlog role surely will feed the sprint role actions (in that way we also reduce the miscommunication from one role to the other one).

**Comment by choldgraf** [comment]
@damianavila yeah I think that is reasonable, and also in a sense it is realistic since we have a small team. I think there's still value in defining the roles separately, just so it's explicit what kind of work needs to be done. Does that make sense?

**Comment by damianavila** [comment]
> I think there's still value in defining the roles separately, just so it's explicit what kind of work needs to be done. Does that make sense?

Sure, I was not implying to merge the roles, just ask the person to use 2 hats :wink:
